src/main/java/com/projetogerenciamentocurso/gerenciamentocurso/service/AlunoService.java

```diff
@@ -0,0 +1,101 @@
+package com.projetogerenciamentocurso.gerenciamentocurso.service;
+
+import javax.transaction.Transactional;
+
+import org.springframework.stereotype.Service;
+
+import com.projetogerenciamentocurso.gerenciamentocurso.GerenciamentoCursoApplication;
+import com.projetogerenciamentocurso.gerenciamentocurso.dto.AlunoDto;
+import com.projetogerenciamentocurso.gerenciamentocurso.dtoresposta.AlunoDtoResposta;
+import com.projetogerenciamentocurso.gerenciamentocurso.exceptions.AlunoException;
+import com.projetogerenciamentocurso.gerenciamentocurso.mensageria.Publisher;
+import com.projetogerenciamentocurso.gerenciamentocurso.models.Aluno;
+import com.projetogerenciamentocurso.gerenciamentocurso.repository.AlunoRepository;
+
+import lombok.AllArgsConstructor;
+
+@Service
+@AllArgsConstructor
+@Transactional
+public class AlunoService {
+	
+	private final Publisher publisher;
+	
+	private final AlunoRepository alunoRepository;
+	
+	public AlunoDtoResposta deletarAluno(AlunoDto aluno) {
+		
+		if(aluno!=null) {
+			 
+			alunoRepository.delete(alunoRepository.getOne(aluno.getMatricula()));
+		}else {
+			throw new AlunoException("Aluno não encontrado");
+		}
+		
+		publisher.send(GerenciamentoCursoApplication.EXCHANGE_NAME,
+				GerenciamentoCursoApplication.ROUTING_KEY_ALUNO_DELETAR, aluno);
+		
+		return criarAlunoDtoResposta(aluno);
+	}
+	
+	public AlunoDtoResposta atualizarAluno(AlunoDto aluno) {
+
+		Aluno alunoModel = alunoRepository.getOne(aluno.getMatricula());
+		if (alunoModel != null) {
+
+			alunoModel.setCpf(aluno.getCpf());
+			alunoModel.setEmail(aluno.getEmail());
+			alunoModel.setFormaIngresso(aluno.getFormaIngresso());
+			alunoModel.setNome(aluno.getNome());
+			alunoModel.setTurma(aluno.getTurma());
+			
+			alunoRepository.save(alunoModel);
+		}else {
+			throw new AlunoException("Aluno não encontrado");
+		}

```

Comment:
Aplicar a sugestão abaixo em todos os services que utilizam o getOne do Spring Data JPA, pois quando a entidade não é encontrada na base de dados é estourada a exceção "EntityNotFoundException"

```suggestion
try {
			Aluno alunoModel = alunoRepository.getOne(aluno.getMatricula());
			
			alunoModel.setCpf(aluno.getCpf());
			alunoModel.setEmail(aluno.getEmail());
			alunoModel.setFormaIngresso(aluno.getFormaIngresso());
			alunoModel.setNome(aluno.getNome());
			alunoModel.setTurma(aluno.getTurma());
			
			alunoRepository.save(alunoModel);
		} catch (EntityNotFoundException e) {
			throw new AlunoException("Aluno não encontrado");
		}
```

Reply:
feito

Reply:
A correção pedida não foi aplicada!

Reply:
feito